source/speech/extensions.py

```diff
@@ -14,3 +14,29 @@
 Notifies when speech is canceled.
 Handlers are called without arguments.
 """
+
+pre_speechCanceled = Action()
+"""
+Notifies when speech is about to be canceled.
+
+@param clearBrailleRegions: wether the braille regions should be cleared in speech output braille mode.
+
+@type clearBrailleRegions: bool
+"""
+
+pre_speech = Action()
+"""
+Notifies when code attempts to speak text.
+
+@param speechSequence: the sequence of text and L{SpeechCommand} objects to speak
+
```

Comment:
I'd suggest grouping param and type for readibility

```suggestion
@param speechSequence: the sequence of text and L{SpeechCommand} objects to speak
```